imagery/i.atcorr/main.cpp

```diff
@@ -204,7 +204,19 @@ class TICache {
 private:
     struct RBitem set_alt_vis(double alt, double vis)
     {
-        struct RBitem rbitem;
+        struct RBitem rbitem = {0,
```

Comment:
Instead of having a lot of 0, 0.0 could we have a function to initialize this struct?

Reply:
The struct has two nested structs.
So i could just think of simple way in which it could be initialize.
No idea how it could be used with function

Reply:
> Instead of having a lot of 0, 0.0 could we have a function to initialize this struct?

This _is_ the function to initialize the struct.

A perhaps more clearer solution could be (diff against main):

```diff
diff --git a/imagery/i.atcorr/main.cpp b/imagery/i.atcorr/main.cpp
index 8b2a1779fa..fc37af791b 100644
--- a/imagery/i.atcorr/main.cpp
+++ b/imagery/i.atcorr/main.cpp
@@ -204,11 +204,22 @@ class TICache {
 private:
     struct RBitem set_alt_vis(double alt, double vis)
     {
-        struct RBitem rbitem;
-
         /* alt and vis must be in meters */
-        rbitem.alt = (alt < 0 ? (int)(alt - 0.5) : (int)(alt + 0.5));
-        rbitem.vis = (int)(vis + 0.5);
+        struct RBitem rbitem = {
+            .alt = (alt < 0 ? (int)(alt - 0.5) : (int)(alt + 0.5)),
+            .vis = (int)(vis + 0.5),
+            .ti = {
+                .asol = 0.0,
+                .ainr = {{0.0, 0.0, 0.0}, {0.0, 0.0, 0.0}},
+                .sb = 0.0,
+                .seb = 0.0,
+                .tgasm = 0.0,
+                .sutott = 0.0,
+                .sdtott = 0.0,
+                .sast = 0.0,
+                .srotot = 0.0,
+                .xmus = 0.0,
+            }};
 
         return rbitem;
     }
```

Reply:
This shorter version seem to satisfy cppcheck:
```diff
diff --git a/imagery/i.atcorr/main.cpp b/imagery/i.atcorr/main.cpp
index 8b2a1779fa..532ec58456 100644
--- a/imagery/i.atcorr/main.cpp
+++ b/imagery/i.atcorr/main.cpp
@@ -204,11 +204,11 @@ class TICache {
 private:
     struct RBitem set_alt_vis(double alt, double vis)
     {
-        struct RBitem rbitem;
-
         /* alt and vis must be in meters */
-        rbitem.alt = (alt < 0 ? (int)(alt - 0.5) : (int)(alt + 0.5));
-        rbitem.vis = (int)(vis + 0.5);
+        struct RBitem rbitem = {
+            .alt = (alt < 0 ? (int)(alt - 0.5) : (int)(alt + 0.5)),
+            .vis = (int)(vis + 0.5),
+        };
 
         return rbitem;
     }
```

Reply:
i have done the recommended changes

Reply:
Thanks :)

Reply:
> Thanks :)



> This shorter version seem to satisfy cppcheck:
> 
> ```diff
> diff --git a/imagery/i.atcorr/main.cpp b/imagery/i.atcorr/main.cpp
> index 8b2a1779fa..532ec58456 100644
> --- a/imagery/i.atcorr/main.cpp
> +++ b/imagery/i.atcorr/main.cpp
> @@ -204,11 +204,11 @@ class TICache {
>  private:
>      struct RBitem set_alt_vis(double alt, double vis)
>      {
> -        struct RBitem rbitem;
> -
>          /* alt and vis must be in meters */
> -        rbitem.alt = (alt < 0 ? (int)(alt - 0.5) : (int)(alt + 0.5));
> -        rbitem.vis = (int)(vis + 0.5);
> +        struct RBitem rbitem = {
> +            .alt = (alt < 0 ? (int)(alt - 0.5) : (int)(alt + 0.5)),
> +            .vis = (int)(vis + 0.5),
> +        };
>  
>          return rbitem;
>      }
> ```


with this code the build failed

Reply:
i did the changes it worked.
Also thanks for giving the version info.

Reply:
> (Current GRASS GIS language standard of C11 and C++11 is declared in https://github.com/OSGeo/grass/blob/main/doc/development/rfc/language_standards_support.md).

During the last community sprint, it was decided to change the checks to a minimum of c++17, and changed in https://github.com/OSGeo/grass/pull/3846

The old RFC was written in a way that it was a fixed version, and couldn't evolve. So, in the same spirit as when it was written and adopted (and the same way that we follow Python versions now), it was chosen to move the range. Adding back checks for newer language standards versions was postponed until we have Ubuntu 24.04 CI ready, in order to use newer compiler versions available (without a custom install) on the distro of users.

The C++17 was also chosen because of it aligns with one of our required dependencies, gdal, which has that standard version for as the lowest supported for their 3.9 version. That standard version only applies for our main branch, that is the 8.5 release that is planned to be out in a long time in the future.

Reply:
> > (Current GRASS GIS language standard of C11 and C++11 is declared in https://github.com/OSGeo/grass/blob/main/doc/development/rfc/language_standards_support.md).
> 
> During the last community sprint, it was decided to change the checks to a minimum of c++17, and changed in #3846
> 
> The old RFC was written in a way that it was a fixed version, and couldn't evolve. So, in the same spirit as when it was written and adopted (and the same way that we follow Python versions now), it was chosen to move the range. Adding back checks for newer language standards versions was postponed until we have Ubuntu 24.04 CI ready, in order to use newer compiler versions available (without a custom install) on the distro of users.
> 
> The C++17 was also chosen because of it aligns with one of our required dependencies, gdal, which has that standard version for as the lowest supported for their 3.9 version. That standard version only applies for our main branch, that is the 8.5 release that is planned to be out in a long time in the future.

Thanks, I was just about to file an issue to bump to C++17 support for future releases :-).

Reply:
Is it something doable to edit/upgrade the RFC instead? To make it evolvable? How long does the approval process take?

Reply:
> Is it something doable to edit/upgrade the RFC instead? To make it evolvable? How long does the approval process take?

We will probably have to make a new (superseding) RFC regarding C++ standard support. RFC 7 is voted for and accepted.